### PR TITLE
Source Google Analytics (UA): specify date format in spec

### PR DESCRIPTION
**Pull request by arsenlosenko**
## What
Resolve issue described here:
https://github.com/airbytehq/oncall/issues/2034
Right now user can enter almost any value for start_date during connector setup, added `format` to spec so only dates could be accepted

## How
Updated spec, added `format` entry to enable datepicker in UI. Also updated `pattern` in spec to make these changes backwards compatible and pass CAT test which generates different fake configs.

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-google-analytics-v4 test report (commit [`3b8f9cd4b1`](https://github.com/airbytehq/airbyte/pull/28149/commits/3b8f9cd4b12b14ead0388f6787b5b86a834bdfc2)) - ✅

⏲️  Total pipeline duration: 24mn08s 

| Step                                                                              | Result   |
|:----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-google-analytics-v4/metadata.yaml | ✅        |
| Connector version semver check                                                    | ✅        |
| Connector version increment check                                                 | ✅        |
| QA checks                                                                         | ✅        |
| Code format checks                                                                | ✅        |
| Connector package install                                                         | ✅        |
| Build source-google-analytics-v4 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                        | ✅        |
| Acceptance tests                                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-google-analytics-v4-format-date-spec/1689084434/3b8f9cd4b12b14ead0388f6787b5b86a834bdfc2/source-google-analytics-v4/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-v4 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-google-analytics-v4 test report (commit [`3b8f9cd4b1`](https://github.com/airbytehq/airbyte/pull/28149/commits/3b8f9cd4b12b14ead0388f6787b5b86a834bdfc2)) - ❌

⏲️  Total pipeline duration: 06mn27s 

| Step                                                                              | Result   |
|:----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-google-analytics-v4/metadata.yaml | ✅        |
| Connector version semver check                                                    | ✅        |
| Connector version increment check                                                 | ✅        |
| QA checks                                                                         | ✅        |
| Code format checks                                                                | ✅        |
| Connector package install                                                         | ✅        |
| Build source-google-analytics-v4 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                        | ✅        |
| Acceptance tests                                                                  | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-google-analytics-v4-format-date-spec/1689088881/3b8f9cd4b12b14ead0388f6787b5b86a834bdfc2/source-google-analytics-v4/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-v4 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-google-analytics-v4 test report (commit [`d61e4f8c58`](https://github.com/airbytehq/airbyte/pull/28149/commits/d61e4f8c585bfb57f8f8aefbf4684a009b924062)) - ✅

⏲️  Total pipeline duration: 25mn07s 

| Step                                                                              | Result   |
|:----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-google-analytics-v4/metadata.yaml | ✅        |
| Connector version semver check                                                    | ✅        |
| Connector version increment check                                                 | ✅        |
| QA checks                                                                         | ✅        |
| Code format checks                                                                | ✅        |
| Connector package install                                                         | ✅        |
| Build source-google-analytics-v4 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                        | ✅        |
| Acceptance tests                                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-google-analytics-v4-format-date-spec/1689246335/d61e4f8c585bfb57f8f8aefbf4684a009b924062/source-google-analytics-v4/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-v4 test
```

File: docs/integrations/sources/google-analytics-v4.md [hunk 1]
```diff
@@ -190,6 +190,7 @@ Incremental sync is supported only if you add `ga:date` dimension to your custom
 
 | Version | Date       | Pull Request                                             | Subject                                                                                      |
 | :------ | :--------- | :------------------------------------------------------- | :------------------------------------------------------------------------------------------- |
+| 0.2.1   | 2023-07-11 | [28149](https://github.com/airbytehq/airbyte/pull/28149) | Specify date format to support datepicker in UI                                                                         |
```
Comment (lazebnyi) [inline docs/integrations/sources/google-analytics-v4.md:193]: Please delete extra space after `Specify date format to support datepicker in UI`.

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json [hunk 2]
```diff
@@ -81,7 +81,9 @@
         "type": "string",
         "title": "Replication Start Date",
         "description": "The date in the format YYYY-MM-DD. Any data before this date will not be replicated.",
-        "examples": ["2020-06-01"]
+        "examples": ["2020-06-01"],
+        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$|^$|[\\s\\S]+$",
+        "format": "date"
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:86]: Do we have an old configuration validation?

**Comment by octavia-squidington-iii** [comment]
##  source-google-analytics-v4 test report (commit [`d7cc3b1d82`](https://github.com/airbytehq/airbyte/pull/28149/commits/d7cc3b1d823c2144258fc87b10f1250236be6a1a)) - ✅

⏲️  Total pipeline duration: 28mn58s 

| Step                                                                              | Result   |
|:----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-google-analytics-v4/metadata.yaml | ✅        |
| Connector version semver check                                                    | ✅        |
| Connector version increment check                                                 | ✅        |
| QA checks                                                                         | ✅        |
| Code format checks                                                                | ✅        |
| Connector package install                                                         | ✅        |
| Build source-google-analytics-v4 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                        | ✅        |
| Acceptance tests                                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-google-analytics-v4-format-date-spec/1689595600/d7cc3b1d823c2144258fc87b10f1250236be6a1a/source-google-analytics-v4/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-v4 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-google-analytics-v4 test report (commit [`37a9c3efa3`](https://github.com/airbytehq/airbyte/pull/28149/commits/37a9c3efa34b1a770e4297fe97be9f9a3e8ebc89)) - ❌

⏲️  Total pipeline duration: 34mn40s 

| Step                                                                              | Result   |
|:----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-google-analytics-v4/metadata.yaml | ✅        |
| Connector version semver check                                                    | ✅        |
| Connector version increment check                                                 | ✅        |
| QA checks                                                                         | ✅        |
| Code format checks                                                                | ✅        |
| Connector package install                                                         | ✅        |
| Build source-google-analytics-v4 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                        | ✅        |
| Acceptance tests                                                                  | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-google-analytics-v4-format-date-spec/1689668632/37a9c3efa34b1a770e4297fe97be9f9a3e8ebc89/source-google-analytics-v4/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-v4 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-google-analytics-v4 test report (commit [`ccbf9a4f75`](https://github.com/airbytehq/airbyte/pull/28149/commits/ccbf9a4f751fdc4eeb48407540688539946cfb43)) - ✅

⏲️  Total pipeline duration: 33mn46s 

| Step                                                                              | Result   |
|:----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-google-analytics-v4/metadata.yaml | ✅        |
| Connector version semver check                                                    | ✅        |
| Connector version increment check                                                 | ✅        |
| QA checks                                                                         | ✅        |
| Code format checks                                                                | ✅        |
| Connector package install                                                         | ✅        |
| Build source-google-analytics-v4 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                        | ✅        |
| Acceptance tests                                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-google-analytics-v4-format-date-spec/1689754327/ccbf9a4f751fdc4eeb48407540688539946cfb43/source-google-analytics-v4/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-v4 test
```

File: docs/integrations/sources/google-analytics-v4.md (hunk 1 above)
Comment (arsenlosenko) [reply docs/integrations/sources/google-analytics-v4.md]: removed space here

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json (hunk 2 above)
Comment (arsenlosenko) [reply airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:86]: by default, dates that are valid for this source are in YYYY-MM-DD format, this change only adds validation to this exact configuration

**Comment by octavia-squidington-iii** [comment]
##  source-google-analytics-v4 test report (commit [`1a1182b552`](https://github.com/airbytehq/airbyte/pull/28149/commits/1a1182b552ce8acf2ecc075a9588769f329c52fa)) - ✅

⏲️  Total pipeline duration: 21mn31s 

| Step                                                                              | Result   |
|:----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-google-analytics-v4/metadata.yaml | ✅        |
| Connector version semver check                                                    | ✅        |
| Connector version increment check                                                 | ✅        |
| QA checks                                                                         | ✅        |
| Code format checks                                                                | ✅        |
| Connector package install                                                         | ✅        |
| Build source-google-analytics-v4 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                        | ✅        |
| Acceptance tests                                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-google-analytics-v4-format-date-spec/1689757213/1a1182b552ce8acf2ecc075a9588769f329c52fa/source-google-analytics-v4/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-v4 test
```

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**